src/vm/core.zig

```diff
@@ -869,24 +881,56 @@ pub const OperandsResult = struct {
         };
     }
 
+    /// Sets the flag indicating the destination operand was deduced.
+    ///
+    /// # Arguments
+    ///
+    /// - `value`: A boolean value indicating whether the destination operand was deduced.
     pub fn setDst(self: *Self, value: bool) void {
-        self.deduced_operands |= if (value) 1 else 0;
+        self.deduced_operands |= @intFromBool(value);
     }
+
+    /// Sets the flag indicating the first operand was deduced.
+    ///
+    /// # Arguments
+    ///
+    /// - `value`: A boolean value indicating whether the first operand was deduced.
     pub fn setOp0(self: *Self, value: bool) void {
-        self.deduced_operands |= if (value) 1 << 1 else 0 << 1;
+        self.deduced_operands |= @as(u8, @intCast(@intFromBool(value))) << 1;
     }
 
+    /// Sets the flag indicating the second operand was deduced.
+    ///
+    /// # Arguments
+    ///
+    /// - `value`: A boolean value indicating whether the second operand was deduced.
     pub fn setOp1(self: *Self, value: bool) void {
-        self.deduced_operands |= if (value) 1 << 2 else 0 << 2;
+        self.deduced_operands |= @as(u8, @intCast(@intFromBool(value))) << 2;
```

Comment:
why not something like `self.deduced_operands |= if (value) @as(u8, 1) << 2 else @as(u8, 0) << 2;` ?